src/containers/Petitions.js

```diff
@@ -7,8 +7,14 @@ import Petitions from 'components/Petitions';
 import getPetitions from 'selectors/petitions';
 
 const PetitionsContainer = React.createClass({
-  componentDidMount () {
-    this.props.fetchPetitions(this.props);
+  // When the component gets added to the DOM,
+  // fetch Petitions if none are defined (clientside)
+  componentWillMount () {
+    const { petitions } = this.props;
+
+    if (!petitions.length) {
+      this.props.fetchPetitions(this.props);
+    }
```

Comment:
hmm yeah true. but it always re-renders on the client on first load which is not ideal.
can we find a way to determine if the params / filters have changed when we do the filtering story?